### DISCUSSION TITLE
Does go-redis provide an implementation of distributed locks?

### DISCUSSION THREAD

**Discussion by musclechen** [redis/go-redis#1962] [Q&A] [answered]
I want a distributed lock that can achieve CAS, automatic extension (watch dog) and so on. Similar to the function of java's Redisson. I don't know if such an interface already exists for go-redis.

**Comment by vmihailenco** [comment, marked as answer]
You can try https://github.com/bsm/redislock

**Reply by musclechen** [reply]
Thanks.
